### PR TITLE
Implementation of Non-Result Board Methods with Tests

### PR DESCRIPTION
**Pull request by okonekp**
Implemented Board class, ship class pulled into this branch is already reviewed and tested.

### CODE REVIEW

**Comment by AlexLeam** [comment]
The code looks good. In the tests, doesn't just using assert have the potential to crash?

File: src/test/java/cs361/battleships/models/BoardTest.java [hunk 1]
```diff
@@ -3,12 +3,56 @@
...
+        board.placeShip(new Ship("CRUISER"),5,'C',true);
+        assert(board.getShips().size() == 1);
+    }
+
+    @Test
+    public void testSetShips() {
+        Board board = new Board();
+        List<Ship> ships = new ArrayList<Ship>();
+        ships.add(new Ship("BATTLESHIP"));
+        assert(board.getShips().size() == 0);
```
Comment (AlexLeam) [inline src/test/java/cs361/battleships/models/BoardTest.java]: This probably won't pass seeing as how you're adding a ship then checking if it's empty

**Comment by okonekp** [comment]
Changed bounds for the placement checking function to be correct; also removed some print lines in the testing functions that were used for testing the tests.

**Comment by okonekp** [comment]
Administrative note: when this is merged, it will resolve #2

**Pull request merged**